src/addons/transitions/ReactTransitionChildMapping.js

```diff
@@ -27,11 +27,7 @@ var ReactTransitionChildMapping = {
       return children;
     }
 
-    if (__DEV__) {
-      return flattenChildren(children, selfDebugID);
-    }
-
-    return flattenChildren(children);
+    return flattenChildren(children, selfDebugID);
```

Comment:
I will trace it down further to make sure it is, but my understanding was that this debug ID is used by the instrumentation. So I was trying to make sure we are passing it along wherever we can.